### PR TITLE
Allow "check core" command to be run from any Python // Resolve #44

### LINKED ISSUE #44: Allow "check core" command to be run from any Python
Subj.

### PR DESCRIPTION
**Pull request by ShahRustam**

### CODE REVIEW

File: pioinstaller/core.py [hunk 1]
```diff
@@ -161,79 +181,74 @@ def check(dev=False, auto_upgrade=False, version_requirements=None):
...
-    result["core_version"] = get_pio_version(platformio)
+    try:
+        pio_state.update(fetch_platformio_state(python_exe))
+    except:  # pylint:disable=bare-except
+        raise exception.InvalidPlatformIOCore("Could not import PlatformIO module")
+
+    return pio_state
+
+
+def fetch_platformio_state(python_exe):
```
Comment (ivankravets) [inline pioinstaller/core.py:230]: rename to `fetch_python_state`

File: pioinstaller/core.py [hunk 2]
```diff
@@ -161,79 +181,74 @@ def check(dev=False, auto_upgrade=False, version_requirements=None):
...
+    code = """import platform
+import json
+import platformio
 
-    return result
+state = {
+   "core_version": platformio.__version__,
+   "python_version": platform.python_version()
+}
+print(json.dumps(state), end="")
```
Comment (ivankravets) [inline pioinstaller/core.py:239]: remove `end=""`

File: pioinstaller/core.py [hunk 3]
```diff
@@ -141,12 +144,29 @@ def check(dev=False, auto_upgrade=False, version_requirements=None):
...
+        {
+            "core_dir": get_core_dir(),
+            "cache_dir": get_cache_dir(),
+            "penv_dir": penv.get_penv_dir(),
+            "penv_bin_dir": penv.get_penv_bin_dir(),
+            "platformio_exe": platformio_exe,
+            "installer_version": __version__,
+            "python_exe": python_exe,
+            "system": util.get_systype(),
+            "is_develop_core": dev,
```
Comment (ivankravets) [inline pioinstaller/core.py:166]: Add `python_version`. We should have it now?

Comment (ShahRustam) [reply pioinstaller/core.py:166]: I got it from fetch_python_state.

Comment (ivankravets) [reply pioinstaller/core.py:166]: Cool, use it.

**Pull request merged**